### PR TITLE
Splits star vampires into 'fanged' and 'coiling'

### PR DESCRIPTION
**Pull request by carlarctg**
#### Summary
Balance "Splits star vampires into a fast-blood-sucking and a strong-grabbing version."

#### Purpose of change

Star vampires are really, really cool. I really like them. They're a cool new addition to the game that makes heavy use of some of the newest mechanics to be added and are pants-shittingly scary.

However, I do think they're overtuned. They're invisible, faster than you, drain your blood almost instantly, and have two strong-level grabs.

This means that they're invisible up until they're right next to you, they use their two grabs on you in one turn, making you unable to wield anything, and then they suck you dry while you try to free yourself. Even if you do manage to free yourself, they can grab you again, and if you do somehow manage to kill them or escape, you're usually a walking corpse by that point - even if you remembered to snag a saline kit, it might not be enough when your veins and arteries are mostly pumping air now.

But I don't want to just blandly nerf them. I like that they hoover your blood up. I like that it's hard to struggle to escape from them! In order to keep these things while making them not a instant round-ender, I split them in two. One drains your blood super-fast, the other is very hard to escape from.

#### Describe the solution

Fanged star vampires are the 'original' vampires. They have one grab now, instead of two, and it's a bit weaker, but otherwise they're intact.

Created a copy-from subtype called 'coiling star vampire'. This version has the usual 2 grabs at 50 strength, but its bloodsucking effects are half as powerful as the fanged version.

Both spawn in the usual places, with the weights for them divided by 2 so they aren't more common than intended. Also made sure the mods got updated too.

#### Describe alternatives you've considered

Nerfing star vampire outright. Boring, cheap, uninteresting, dull.

Making the star vampire get full after a certain point. Too much effort and intrudes on the original creator's plans, I think.

#### Testing

Spawned a fanged vampire in, struggled against it and managed to escape and run a few tiles before dying to it.

Spawned a coiling vampire in, got grabbed and couldn't escape while it slowly sucked me dry. All working as intended.

### CODE REVIEW

**Comment by Standing-Storm** [comment]
Hmm.  On the one hand, I persist in my belief that star vampires are fine, but on the other hand, you're not the first person to say they're too strong.

I would say, though, that rather thing splitting the star vampire into two separate monsters, maybe applying a lesser version of both nerfs to the original monster would be better (reduce grab strength to ~40 and blood drain to ~175/475 or so) to avoid giving away the game with the monster name. I've also seen several people say that fighting a star vampire was the first time they were scared in CDDA in a while and I think it would be a shame to dilute that.

**Comment by carlarctg** [comment]
> Hmm. On the one hand, I persist in my belief that star vampires are fine, but on the other hand, you're not the first person to say they're too strong.
> 
> I would say, though, that rather thing splitting the star vampire into two separate monsters, maybe applying a lesser version of both nerfs to the original monster would be better (reduce grab strength to ~40 and blood drain to ~175/475 or so) to avoid giving away the game with the monster name. I've also seen several people say that fighting a star vampire was the first time they were scared in CDDA in a while and I think it would be a shame to dilute that.

That was the intention here. It'd keep the spookiness of the grabs, and the spookiness of the blood sucking, without instantly ending the run. I'd say a nerf to the default vampire's capabilities would be more of a dilution as you're making both attacks less scary rather than separating them into different monsters.

If you strongly believe reducing the stats is a better option, I'll do that.

**Comment by Standing-Storm** [comment]
> If you strongly believe reducing the stats is a better option, I'll do that.

The major problem people have seems to be winning but still dying (as you say above), so reducing the blood drain by a small amount should remove how often that happens while still making them scary--Moderate Hypovolemic Shock is better than dying but still a serious threat (especially if there are other enemies nearby), and a slightly lower grab strength maintains the conceit of multiple tentacles grasping their prey (why I included two grabs in the first place) while allowing stronger targets to escape more easily. 

I don't think it's a bland nerf, and if it leads to star vampires being shrugged off, then we can always make further adjustments in the future, so I would prefer just reducing the stats.

**Comment by I-am-Erk** [comment]
I've never fought these things but might I suggest you keep the two versions, but give them the same name and appearance? This means that a person doesn't know exactly what a star vampire is going to hit like, but they're still not quite as instantly deadly as before. That also removes the slightly too long names.

File: data/json/effects.json [hunk 1]
```diff
@@ -4799,6 +4799,23 @@
       { "vitamin": "redcells", "rate": [ [ -250, -650 ] ], "tick": [ "1 s" ] }
     ]
   },
+  {
+    "id": "star_vampire_blood_drink",
```
Comment (ZhilkinSerg) [inline data/json/effects.json:4803]: There is already an effect with same id, so this entry would overwrite it. This needs to have a different id or previously existing entry should be removed.

**Pull request merged**